presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryExecutor.java

```diff
@@ -65,6 +86,26 @@ public StatementClient startQuery(ClientSession session, String query)
     @Override
     public void close()
     {
+        if (closer.isPresent()) {
+            closer.get().accept(this);
+        }
+        else {
+            closeHttpClient();
+        }
+    }
+
+    int reference()
```

Comment:
remove reference counting. I don't see the reason why you need reference counting here. Such stateful counters are always error prone. It makes sense to avoid it when it is possible